### PR TITLE
Enable controller-runtime metrics for Kubermatic controller manager

### PR DESCRIPTION
**Pull request by alvaroaleman**
**What this PR does / why we need it**:

Enables the controller-runtime metrics inside the Kubermatic controller manager. This aids in fixing #2342 as there is a metric for the number of  requests by method, e.G.

```
rest_client_requests_total{code="200",host="35.198.89.109",method="PUT"} 135
```

Unfortunately this means that we cant use the prometheus default Registry anymore, because controller-runtime registers its metrics in various `init()` calls, meaning that even if we replace its Registry with the Prometheus default one, it then wont have its metrics.

Instead I just started using the `ctrlruntime` registry everywhere

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Documentation**:

**Release note**:

```release-note
none
```

### CODE REVIEW

**Comment by alvaroaleman** [comment]
cc @xrstf because this will get you a lot of shiny new metrics :)

**Comment by alvaroaleman** [comment]
@mrIncompetent Changed the code to instrument both the default Prometheus registry and the ctrlruntimemetrics registry, PTAL

File: api/cmd/kubermatic-controller-manager/metrics.go [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+	ctrlruntimemetrics.Registry.Unregister(prometheus.NewGoCollector())
+}
+
+// We have our own metrics server implementation that gathers the metrics from
+// both the default prometheus registry and the ctrltuntimemetrics registry.
+// The background is that the latter is not configurable at all and we don't
+// want to force developers into using it, because that is counterintuitive
+// and prone to be forgotten
+type metricsServer struct {
+	gatherers     prometheus.Gatherers
```
Comment (mrIncompetent) [inline api/cmd/kubermatic-controller-manager/metrics.go:35]: :scream: 
I didn't saw that. Nice!

**Comment by mrIncompetent** [comment]
This is a beautiful PR!

/lgtm
/approve

**Pull request merged**